onnxruntime/contrib_ops/cuda/bert/attention.cc

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "attention.h"
+#include "core/framework/tensorprotoutils.h"
+#include "attention_impl.h"
+
+using namespace onnxruntime::cuda;
+using namespace ::onnxruntime::common;
+using namespace ONNX_NAMESPACE;
+
+namespace onnxruntime {
+namespace contrib {
+namespace cuda {
+
+#define REGISTER_KERNEL_TYPED(T)                                  \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(                                  \
+      Attention,                                                  \
+      kOnnxDomain,                                                \
+      1,                                                          \
+      T,                                                          \
+      kCudaExecutionProvider,                                     \
+      KernelDefBuilder()                                          \
+          .TypeConstraint("T", DataTypeImpl::GetTensorType<T>()), \
+      Attention);
+
+REGISTER_KERNEL_TYPED(float)
+
+Attention::Attention(const OpKernelInfo& info) : CudaKernel(info) {
+  int64_t numHeads = 0;
```

Comment:
>numHeads [](start = 10, length = 8)

This variable name in ORT coding style would b num_heads

Reply:
I'll update the variable names.